### DISCUSSION TITLE
How to calculate grain boundary plane orientation with obtained Eulers angle?

### DISCUSSION THREAD

**Discussion by kchoi74** [mtex-toolbox/mtex#733] [Ask Anything]
Greetings All!

I have a question for calculating "grain boundary plane orientation" by using Euler angles of the grains.

Are there any codes that I can calculate grain boundary plane orientations of two grains, shown below?

![image](https://user-images.githubusercontent.com/75606041/106221604-4837ee80-6221-11eb-84f5-6541eaa41bf7.png)

Thanks!

**Comment by ralfHielscher** [comment]
Hi,

```matlab
gb = grains.boundary;
t = mean(gb(grains('id',338),grains('id',332)).direction)
```
gives you the mean boundary trace between these two grains. Hence, the 2d boundary normal will be 
```matlab
n2 = rotate(t, 90*degree)
```
Next you need to know the inclination of the boundary normal with respect to the measurement surface - lets assume this is 30*degree.
and you get the 3d normal of the boundary plane
```matlab
n3 = rotation.byAxisAngle(t, 30*degree) * n2;
```
Finally, you can use the grain orientations to transfer this into crystal coordinates, by
```matlab
nc = inv(grains('id',332).meanOrientation) * n3
```

I hope this helps.

Ralf.






If you now know the inclination

**Reply by kchoi74** [reply]
Thanks for the quick reply!

Can I ask you one more question?

Where does the 30*degree come from?

Sorry for asking basics....

Thanks.

**Comment by lfgmorales** [comment]
Hi


The 30° is an assumption on the calculation, this is the tilt angle of your grain boundary in 3D, either you know that (from a cross section along your grain boundary), or you have to assume an angle


cheers



Luiz


Luiz F. G. Morales
ScopeM & Structural Geology and Tectonics Group
ETH Zürich
________________________________

**Reply by kchoi74** [reply]
Hi
Thanks for the reply to me back!

So if I want to get the 'exact GB plane orientation (or tilt angle)', then do I need to take it by TEM?

Is there any way that I can get the exact GB plane orientation through MTEX? 
(I think this is the question of the basic EBSD...)

Thanks,
David.

**Reply by ralfHielscher** [reply]
There is no way to determine the three dimensional GB plane orientation from 2d-EBSD data only. You certainly need additional information. A theoretical approach would be if you know that only specific habit planes are possible.

Ralf.

**Comment by lfgmorales** [comment]
You don’t need to do TEM, but you need to access the third dimension of your grain boundary, one way to do that is via focused ion beam methods

Cheers


Luiz

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 5 Feb 2021, at 10:49, Ralf Hielscher <notifications@github.com> wrote:

﻿

There is no way to determine the three dimensional GB plane orientation from 2d-EBSD data only. You certainly need additional information. A theoretical approach would be if you know that only specific habit planes are possible.

Ralf.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/733#discussioncomment-340878>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN57YIOXIFQGQKADSCDS5O5KDANCNFSM4WYCQC4A>.

[ { "@context": "http://schema.org", "@type": "EmailMessage", "potentialAction": { "@type": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/733#discussioncomment-340878", "url": "https://github.com/mtex-toolbox/mtex/discussions/733#discussioncomment-340878", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { "@type": "Organization", "name": "GitHub", "url": "https://github.com" } } ]
